### PR TITLE
Preserve upper bound for right shift

### PR DESCRIPTION
**Pull request by panacekcz**
This PR fixes kelloggm#214 (upper bounds for right shift) by copying the upper bound annotation from the left operand, if it is non-negative.

After this is merged, kelloggm#217 can be implemented.

### CODE REVIEW

File: checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -406,4 +406,10 @@ private AnnotationMirror checkForMathRandomSpecialCase(Tree randTree, Tree arrLe
         }
         return null;
     }
+
+    /** Checks if the expression is non-negative, i.e. it has Positive on NonNegative annotation. */
```
Comment (kelloggm) [inline checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java]: You should use this method in the modulo PR too.

Comment (kelloggm) [reply checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java]: And, should this replace `isNonNegative` in `LowerBoundTransfer`? At the very least, please leave a TODO there that we can come back to later.

Comment (panacekcz) [reply checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java]: Not exactly - the parameter type is different; perhaps they could two overloads in one class.
I added the TODO.

**Comment by panacekcz** [comment]
Can this be merged?

**Pull request merged**